fdbserver/MoveKeys.actor.cpp

```diff
@@ -2354,16 +2381,36 @@ ACTOR Future<Void> cleanUpDataMove(Database occ,
                                    MoveKeysLock lock,
                                    FlowLock* cleanUpDataMoveParallelismLock,
                                    KeyRange keys,
-                                   const DDEnabledState* ddEnabledState) {
+                                   const DDEnabledState* ddEnabledState,
```

Comment:
I would suggest moving all the new logic back to DDRelocationQueue, and keep the `cleanUpDataMove()` simple, its function is solely to cleanUp the target data move. If the target data move is not found, it can return a `retry_data_move_deletion_marked` error, and DDRelocationQueue::cancelDataMove() should do the delay and retry `cleanUpDataMove()`